app/Filament/Widgets/ServiceRequestLineChart.php

```diff
@@ -0,0 +1,107 @@
+    see <https://github.com/canyongbs/aidingapp/blob/main/LICENSE.>
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace App\Filament\Widgets;
+
+use Carbon\Carbon;
+use Filament\Widgets\ChartWidget;
+use AidingApp\ServiceManagement\Models\ServiceRequest;
+
+class ServiceRequestLineChart extends ChartWidget
+{
+    protected static ?string $heading = 'Service Requests in last 30 days';
+
+    protected static ?string $maxHeight = '300px';
+
+    protected int | string | array $columnSpan = [
+        'sm' => 2,
+        'md' => 3,
+        'lg' => 3,
+    ];
+
+    protected function getOptions(): array
+    {
+        return [
+            'scales' => [
+                'y' => [
+                    'beginAtZero' => true,
+                    'ticks' => [
+                        'stepSize' => 0,
+                        'beginAtZero' => true,
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    protected function getData(): array
+    {
+        $totalCreatedPerDay = ServiceRequest::query()
+            ->toBase()
+            ->selectRaw('DATE(created_at) as date')
+            ->selectRaw('COUNT(*) as total')
+            ->where('created_at', '>', now()->subDays(30))
+            ->groupBy('date')
+            ->orderBy('date')
+            ->pluck('total', 'date');
+
+        $serviceRequestTotal = 0;
+        $serviceRequestTotalPerDay = [];
+
+        foreach (range(29, 0) as $daysAgo) {
+            $date = Carbon::now()->subDays($daysAgo)->toDateString();
+
+            $serviceRequestTotal += $totalCreatedPerDay[$date] ?? 0;
```

Comment:
@ankit-canyon We'll still need to make an adjustment here so that we are not aggregating the total count, but showing exactly how many records were created on each day.